### ISSUE TITLE
New aravis-5 branch created to use latest version of aravis

### ISSUE THREAD

**Issue by MarkRivers** [areaDetector/aravisGigE#40]
I created a new aravis-5 branch.  It downloads and builds aravis 0.5.13, which is the latest version.  I have it successfully built and running a Point Grey BlackflyS camera.

The changes were pretty minor.

Makefiles
  - The library and include directory change from aravis-0.4 to aravis-0.6.  I would have expected them to be aravis-0.5, but they are 0.6.
  - Added usb-1.0 to the IOC Makefile, because I built aravis with USB camera support

aravisCamera.cpp
 - Changed ARAVIS_VERSION to 0.5.13
  - Automatically determine packet size with call to arv_gv_device_auto_packet_size().  Previously it was setting a default packet size of ARV_GV_DEVICE_GVSP_PACKET_SIZE_DEFAULT, but that  is no longer defined.
-  Call arv_stream_set_emit_signals(stream,FALSE) before unreferencing stream, as instructed in a warning I saw when running without doing this.

install.sh
  - Changed tar file downloaded from aravis repository on ftp.gnome.org.
  - Commented out applying the patch because it did not work, and does not seem to be needed?

**Comment by MarkRivers** [comment]
I just ran arv-tool-0.6 on one of our Linux servers that has a Point Grey Grasshopper3 USB3 camera connected.  arv-tool found the camera, which is a good sign.
```
Viper:devel/areaDetector/aravisGigE>bin/linux-x86_64/arv-tool-0.6
PointGrey-010AA71D (USB3)
Allied Vision Technologies-02-2142A-06178 (164.54.160.58)
Allied Vision Technologies-02-2604A-07008 (164.54.160.104)
Allied Vision Technologies-50-0503317598 (164.54.160.62)
PointGrey-17165235 (164.54.160.215)
Prosilica-02-2142A-06110 (164.54.160.57)
```
It will be interesting to see if aravis can replace FlyCap2 for controlling these cameras.  The problem with FlyCap2 is that versions more recent than 2.9.3 cannot run on RHEL7 or Centos7 because they require a newer version of GLIBC.  Point Grey only officially supports Ubuntu 16 for new releases of FlyCap2, and for any release of their new Spinnaker SDK.  The current version of FlyCap2 is 2.12.3, so those of us with Centos7 are falling behind and cannot update.

**Comment by thomascobb** [comment]
That would be interesting, I'm not sure how much GigE specific code there is in aravisGigE, but it might be relatively simple to add support for USB vision cameras to it...

**Comment by MarkRivers** [comment]
I hooked up my Point Grey Grasshopper3 USB-3 camera (model GS3_U3_23S6M-C).

I did the same thing I would do for a new GigE camera, i.e. run arv-tool to generate the XML file, run makeDbAndEdl.py to generate the database and edm screens, run makeAdl.py to generate the medm screens.  Add the camera type to aravisTop.adl, and add the camera to st.cmd.

It works with no changes to the driver!  I am streaming 1920x1200 at 82 frames/s.  There are some error messages because it is trying to read GigE statistics I think, but they are not preventing it from operating.  This is really nice, since it frees us from the Point Grey SDKs for USB cameras as well.

I had hoped to test the 10Gbit Oryx camera with aravis today, but the GigE card I bought is flaky.  I should be able to test next week when I get back from vacation.

**Comment by MarkRivers** [comment]
I have made some changes to aravisCamera.cpp so that it only calls GigE specific functions (arv_gv_*) if ARV_IS_GV_DEVICE() or ARV_IS_GV_STREAM() is true.  This eliminates error messages when using USB-3 cameras.

I was able to get 140 frames/s for my 1920x1200 camera, which is about 307 MB/s.  This is less than the 163 frames/s rating for the camera, which I have been able to achieve on Windows and on other Linux systems using the ADPointGrey driver.  On this Linux system I briefly got 163 frames/s with the ADPointGrey driver, but then it fell back to about 140, so ADPointGrey and aravisGigE seem to be very similar.
